docs/guide/custom-functions.md

```diff
@@ -140,16 +230,81 @@ the calculations.
 // arguments here are displayed just to show the structure
 public hyper(ast, state) {
     return 'Hyperformula'.length;
+  };
+```
+
+### `runFunction()`
+
+Wrap your custom function in the built-in `runFunction()` method.
+
+The `runFunction()` method wraps your function's calculations and:
+- Validates the [optional parameter](#optional-parameters) settings
+- Validates the arguments against your [argument validation options](#argument-validation-options)
+- Checks if values returned by your function are in the right format
+
+```javascript
+import { FunctionPlugin } from 'hyperformula';
+
+export class CountHF extends FunctionPlugin {
+
+  public static implementedFunctions = {
+    'HYPER': {
+      method: 'hyper',
+      // set your optional parameters
+      arrayFunction: false,
+      // set your argument validation options
+      parameters: {
+        passSubtype: false
+      }
+    }
+  };
+
+  public hyper(ast, state) {
+    return this.runFunction(() => 'Hyperformula'.length)
   }
 };
 ```
 
+### Returning errors
+
+If you want your custom function to return an error, check the [API reference](../api) for the HyperFormula [error types](types-of-errors.md).
+
+:::tip
+All HyperFormula [error types](types-of-errors.md) support optional custom error messages. Put them to good use: let your users know what caused the error and how to avoid it in the future.
+:::
+
+For example, if you want to return a `#DIV/0!` error with your custom error message, use the [`CellError` class](../api/classes/cellerror.md), and the [DIV_BY_ZERO](../api/enums/errortype.md#div-by-zero) error type:
+
+```javascript
+// import `CellError` and `ErrorType`
+import { FunctionPlugin, CellError, ErrorType } from "hyperformula";
+
+public hyper({ args }) {
+    if (!args.length) {
+      // create a `CellError` instance with an `ErrorType` of `DIV_BY_ZERO`
+      // with your custom error message (optional)
+      return new CellError(ErrorType.DIV_BY_ZERO, 'Sorry, cannot divide by zero!');
+    }
+    
+    else {
+      return 'Hyperformula'.length;
+    }
```

Comment:
`else` is not necessary here as we return error above. Aloe we've lost `runFunction` from the previous step
```suggestion
      return this.runFunction(() => 'Hyperformula'.length)
```

Reply:
@wojciechczerniak - done: f4313f1